docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,29 @@
+# Contribution Guidelines
+
+## IDE
+
+IntelliJ IDEA Community / Ultimate. (Please report any issues with Community as I'm using Ultimate.)
+
+Recommended plugins:
+
+* detekt (config file: [config/detekt/detekt.yml](../config/detekt/detekt.yml))
+  For a faster feedback loop.
+* Markdown
+  Syntax highlight of .md files, and rule descriptions.
+
+## Local development
+
+It is recommended to run `gradlew build` before `git push`, as CI will do the same.
+
+## Code style
+
+Notable things:
+
+* The project uses TABs as indent, except for YAML.
+* CI / `gradlew build` will tell you off for most things, so reviews don't have to.
+* The `.idea` folder is version controlled, this means when you open the project it should pick up:
+  * Code styles specific to the project
+  * Inspection profile specific to the project
+
+  With these, it is recommended to write code without any warnings.
```

Comment:
Isn't `-Werror` specified? So not really a "recommendation"?

```
e: warnings found and -Werror specified
w: file:///Users/timsearle/Developer/open-source/net.twisterrob.ghlint/modules/ghlint-rules/src/test/kotlin/net/twisterrob/ghlint/rules/ExplicitJobPermissionsRuleTest.kt:259:8 Variable 'unused' is never used
```

Reply:
I tried running IDEA on CI, but it was too slow / reporting required a third party.

Reply:
I'll see if I can raise severity in IDEA for these for a tighter loop.

Reply:
Well, no. https://youtrack.jetbrains.com/issue/KTIJ-16488